src/store/slices/myAccommodationsListSlice.ts

```diff
@@ -0,0 +1,76 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { Accommodation, StateType } from 'src/types';
+
+import { onError, onPending } from '../stateResults';
+import { AsyncThunks } from '../thunks';
+
+type MyAccommodationsStateType = StateType<Accommodation[]>;
+
+const initialState: MyAccommodationsStateType = {
+  error: null,
+  pending: false,
+  result: null,
+};
+
+const myAccommodationsListSlice = createSlice({
+  name: 'myAccommodationsList',
+  initialState,
+  reducers: {
+    reset: () => initialState,
+
+    clearError: (state) => {
+      state.error = null;
+    },
+
+    createAccommodation: (state, action) => {
+      const createAccommodation = action.payload.data;
+      state.result?.push(createAccommodation);
```

Comment:
Would it be better to create array with 1 element instead of ignoring creation in case result equals null?

Reply:
state.result = state.result || []

  state.result.push(createAccommodation)
you mean like this?

Reply:
yes